apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/LoadedFileFilter.java

```diff
@@ -0,0 +1,134 @@
+package gov.cms.bfd.server.war.stu3.providers;
+
+import ca.uhn.fhir.rest.param.DateParam;
+import ca.uhn.fhir.rest.param.DateRangeParam;
+import com.google.common.base.Charsets;
+import com.google.common.hash.BloomFilter;
+import com.google.common.hash.Funnels;
+import java.util.Date;
+
+/**
+ * LoadedFile filters are used to determine if a given beneficiary was updated in particular loaded
+ * file. Beneath the covers, they use BloomFilters (see <a
+ * href="https://en.wikipedia.org/wiki/Bloom_filter">Bloom Filters</a>) which are space efficient.
+ */
+public class LoadedFileFilter {
```

Comment:
What are the trade offs between the bloomfilter implementation and a optimized date range query (as I understand it they can be very cheap if the correct indexes are in place because of ordering).

Reply:
answering my own question: indexing becomes too large for some of the large tables.